packages/transaction-controller/src/types.ts

```diff
@@ -490,6 +490,39 @@ export type TransactionMeta = {
   };
 };
 
+export type TransactionBatchMeta = {
+  /**
+   * Network code as per EIP-155 for this transaction.
+   */
+  chainId: Hex;
+
+  /**
+   * ID of the associated transaction batch.
+   */
+  id: string;
+
+  /**
+   * Data for any nested transactions.
+   * For example, in an atomic batch transaction via EIP-7702.
+   */
+  transactions?: NestedTransactionMetadata[];
+
+  /**
+   * The ID of the network client used by the transaction.
+   */
+  networkClientId: NetworkClientId;
+
+  /**
+   * Origin this transaction was sent from.
+   */
+  origin?: string;
+
+  /**
+   * Timestamp associated with this transaction.
+   */
+  time: number;
```

Comment:
Do we need this at this stage?

Reply:
I've only added this property to use in the trim function. Now that we have removed that function, I'm also removing this property for simplicity.